src/lib/components/Button/Button.jsx

```diff
@@ -6,31 +6,32 @@ import { getRootColorClassName } from '../_helpers/getRootColorClassName';
 import { getRootSizeClassName } from '../_helpers/getRootSizeClassName';
 import { resolveContextOrProp } from '../_helpers/resolveContextOrProp';
 import { transferProps } from '../_helpers/transferProps';
-import withForwardedRef from '../withForwardedRef';
 import { ButtonGroupContext } from '../ButtonGroup';
 import getRootLabelVisibilityClassName from './helpers/getRootLabelVisibilityClassName';
 import getRootPriorityClassName from './helpers/getRootPriorityClassName';
 import styles from './Button.scss';
 
-export const Button = ({
-  afterLabel,
-  beforeLabel,
-  block,
-  disabled,
-  endCorner,
-  feedbackIcon,
-  forwardedRef,
-  id,
-  label,
-  labelVisibility,
-  priority,
-  size,
-  startCorner,
-  type,
-  color,
-  ...restProps
-}) => {
+export const Button = React.forwardRef((props, ref) => {
+  const {
+    afterLabel,
+    beforeLabel,
+    block,
+    disabled,
+    endCorner,
+    feedbackIcon,
+    id,
+    label,
+    labelVisibility,
+    priority,
+    size,
+    startCorner,
+    type,
+    color,
+    ...restProps
+  } = props;
+
   const context = useContext(ButtonGroupContext);
+  const buttonRef = ref || React.createRef();
```

Comment:
I have a question. Why don't you use `ref` directly. We don't want to create reference which won't be used. If ref is not required, we should just omit it.

Reply:
I agree with @bedrich-schindler I think it is OK to let iot default to `undefined`